subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoReportAggregationPlugin.java

```diff
@@ -140,4 +145,23 @@ private void configureReportTaskInputs(JacocoReport task, ArtifactView classDire
         task.getClassDirectories().from(classDirectories.getFiles());
         task.getSourceDirectories().from(sourceDirectories.getFiles());
     }
+
+    /**
+     * Read the first few magic bytes of the file and compare them to known zip signatures to determine if it is an archive.
+     *
+     * @see <a href="https://en.wikipedia.org/wiki/List_of_file_signatures">List of file signatures</a>
+     * @param f the file to check
+     * @return {@code true} if the file is an archive, {@code false} otherwise
+     */
+    private static boolean isArchive(File f) {
+        if (f.exists()) {
+            try (RandomAccessFile raf = new RandomAccessFile(f, "r")) {
+                return ZIP_SIGNATURES.contains(raf.readInt());
+            } catch (IOException e) {
+                throw new RuntimeException("Error checking file signature", e);
+            }
+        } else {
+            return false;
+        }
+    }
```

Comment:
This is very cool, but would it be enough to instead filter files ending with `.exec`?

Reply:
I didn't want to rely on fille extensions in case the file got renamed, or the task's defaults changed.  And while it would be better to positively ID exec files rather than exclude (some) non-exec files,  there's no info on that exec file layout anywhere I could find.  So I feel like we have to treat its format as an implementation detail subject to change.